pkg/virt-launcher/virtwrap/network/ndp/connection.go

```diff
@@ -74,6 +75,15 @@ func NewNDPConnection(ifaceName string) (*NDPConnection, error) {
 		return nil, fmt.Errorf("failed to join %s multicast group: %v", routersMulticastGroup.String(), err)
 	}
 
+	icmpListenerFD, err := icmpListener.File()
+	if err != nil {
+		return nil, err
+	}
+	if err := syscall.SetsockoptString(int(icmpListenerFD.Fd()), syscall.SOL_SOCKET, syscall.SO_BINDTODEVICE, ifaceName); err != nil {
```

Comment:
I think the setting doesn't have the desired effect since they are done after the `ListenIP`.
https://github.com/golang/go/issues/9661
> There are some cases[1] where setting socket options (like SO_REUSEPORT) is just unavoidable. In most cases, we > can pull out the conn.File() and set them. But those few annoying cases where the option must be set on the socket > > before listening or dialing are impossible to handle with today's net package. (please tell me i'm wrong!)

Please try to use listenConfig.ListenPacket("ip6:ipv6-icmp", listenAddr)
e.g - 
```
lc := CreateListenConfig()
l, err := lc.ListenPacket(context.Background(), "ip6:ipv6-icmp", listenAddr)
```

```
func CreateListenConfig(ifaceName string) net.ListenConfig {
	return net.ListenConfig{
		Control: func(network, address string, c syscall.RawConn) error {
			var opErr error
			err := c.Control(func(fd uintptr) {
				opErr = syscall.SetsockoptInt(int(fd), syscall.SOL_SOCKET, syscall.SO_BINDTODEVICE, ifaceName)
			})
			if err != nil {
				return err
			}
			return opErr
		},
	}
}
```
Reference - 
https://github.com/kubevirt/kubevirt/blob/master/pkg/virt-launcher/virtwrap/network/dhcp/socket_listener.go#L55